Source/Particles/Collision/BinaryCollision.H

```diff
@@ -291,22 +503,49 @@ public:
                     auto dV = MathConst::pi*(2.0_rt*ri+1.0_rt)*dr*dr*dz;
 #endif
                     // Call the function in order to perform collisions
+                    // If there are product species, p_mask, p_pair_indices_1/2, and
+                    // p_pair_reaction_weight are filled here
                     binary_collision_functor(
                         cell_start_1, cell_stop_1, cell_start_2, cell_stop_2,
                         indices_1, indices_2,
                         soa_1, soa_2, get_position_1, get_position_2,
-                        q1, q2, m1, m2, dt*ndt, dV, engine );
+                        q1, q2, m1, m2, dt*ndt, dV,
+                        cell_start_pair, p_mask, p_pair_indices_1, p_pair_indices_2,
+                        p_pair_reaction_weight, engine );
                 }
             );
+
+
+            // Create the new product particles and define their initial values
+            // num_added: how many particles of each product species have been created
+            const amrex::Vector<int> num_added = m_copy_transform_functor(soa_1, soa_2,
+                                                    soa_products_data,
+                                                    get_position_1, get_position_2,
+                                                    get_position_products_data,
+                                                    p_mask, products_np_data,
+                                                    copy_species1, copy_species2,
+                                                    p_pair_indices_1, p_pair_indices_2,
+                                                    p_pair_reaction_weight);
+
+            for (int i = 0; i < n_product_species; i++)
+            {
+                ParticleTileType& ptile_product = product_species_vector[i]->ParticlesAt(lev, mfi);
+                setNewParticleIDs(ptile_product, products_np[i], num_added[i]);
```

Comment:
Also, this function was protected so I had to make the `BinaryCollision` class a friend of the `WarpXParticleContainer` class. Maybe it would be simpler to make this function public?

Reply:
Thanks! In fact, it looks like I was mistaken, it's the `defineAllParticleTiles` function that was protected, not this one. But anyways, I went ahead and made `defineAllParticleTiles` public in the latest commit so that I don't have to make `BinaryCollision` a friend of `WarpXParticleContainer`.